homeassistant/components/advantage_air/cover.py

```diff
@@ -113,3 +123,34 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
                     }
                 }
             )
+
+
+class AdvantageAirThingCover(AdvantageAirThingEntity, CoverEntity):
+    """Representation of Advantage Air Cover controlled by MyPlace."""
+
+    def __init__(self, instance, thing, device_class):
+        """Initialize an Advantage Air Things Cover."""
+        super().__init__(instance, thing)
+        self._attr_device_class = device_class
+
+    @property
+    def is_closed(self) -> bool:
+        """Return if cover is fully closed."""
+        return self._data["value"] == 0
+
+    @property
+    def current_cover_position(self) -> int:
+        """Return covers current position as a percentage."""
+        return self._data["value"]
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Fully open zone vent."""
+        await self.set({"id": self._id, "value": 100})
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Fully close zone vent."""
+        await self.set({"id": self._id, "value": 0})
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Change cover position."""
+        await self.set({"id": self._id, "value": kwargs[ATTR_POSITION]})
```

Comment:
Looks like this might be missing some supported features

https://developers.home-assistant.io/docs/core/entity/cover?_highlight=cover#supported-features

Reply:
Good catch, I double checked the documentation and set_cover_position shouldn't actually be supported, so I have removed the method and set `_attr_supported_features` correctly.